subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/BuildOperationNotificationsFixture.groovy

```diff
@@ -1,131 +0,0 @@
-/*
- * Copyright 2017 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.gradle.integtests.fixtures
-
-import groovy.json.JsonSlurper
-import org.gradle.api.execution.internal.TaskOperationDetails
-import org.gradle.integtests.fixtures.executer.GradleExecuter
-import org.gradle.integtests.fixtures.executer.InitScriptExecuterFixture
-import org.gradle.internal.operations.notify.BuildOperationFinishedNotification
-import org.gradle.internal.operations.notify.BuildOperationNotificationListener
-import org.gradle.internal.operations.notify.BuildOperationNotificationListenerRegistrar
-import org.gradle.internal.operations.notify.BuildOperationStartedNotification
-import org.gradle.test.fixtures.file.TestDirectoryProvider
-import org.gradle.test.fixtures.file.TestFile
-import org.gradle.util.TextUtil
-
-import java.util.concurrent.ConcurrentHashMap
-
-class BuildOperationNotificationsFixture extends InitScriptExecuterFixture {
-
-    private final TestFile operationsDir
-    private final Map<Object, CompleteBuildOperation> operations = [:]
-
-    BuildOperationNotificationsFixture(GradleExecuter executer, TestDirectoryProvider projectDir) {
-        super(executer, projectDir)
-        this.operationsDir = projectDir.testDirectory.file("operations")
-    }
-
-    @Override
-    String initScriptContent() {
```

Comment:
> The implementation can be cleaned up considerably. 

Please do so. This is some of our most important infrastructure, so we need it to be very clear.

> I had to get it working very quickly.

We have another week until RC1. Even if we didn't, I'd rather postpone RC1 than rush changes to central infrastructure.